data/datasets/poetry_instruction/prepare.py

```diff
@@ -0,0 +1,151 @@
+    "This is a poem about $topic that I just wrote. It's called $title: \n$poem",
+    "Here's a poem I composed about $topic. It's called $title: \n$poem",
+]
+
+replies_notTopic = [
+    "Here's a poem: \n$title\n$poem",
+    "Sure, I can do that. Here's a poem. I call it $title: \n$poem",
+    "Okay, a poem: \n$title\n$poem",
+    "Of course! It's called $title: \n$poem",
+    "It's called $title: \n$poem",
+    "Here's your poem: \n$title\n$poem",
+    "I've written a poem for you. The title is $title: \n$poem",
+    "Here's a beautiful poem for you. It's called $title: \n$poem",
+    "This is a poem that I just wrote. It's called $title: \n$poem",
+    "Here's a poem I composed. It's called $title: \n$poem",
+]
+
+# "titling_prompts" are for tasks requesting that the assistant titles a poem. They make up 5% of the dataset.
+titling_prompts = [
+    "Title this poem: \n$poem",
+    "Come up with a unique title for my poem: \n$poem",
+    "What should I call this poem? \n$poem",
+    "Name this poem: \n$poem",
+    "What would be a good title for this poem? \n$poem",
+    "I need help coming up with a title for my poem. \n$poem",
+    "$poem\nWhat should I call this poem?",
+]
+
+titling_replies = [
+    "Based on the poem, a good title could be $title.",
+    "I suggest titling this poem $title.",
+    "How about calling it $title?",
+    "You could name this poem $title.",
+    "The title that comes to mind is $title.",
+    "Perhaps $title would be a fitting title for this poem.",
+    "I think $title would be a great title for this poem.",
+    "This poem seems like it could be called $title to me.",
+    "$title is a good title for this poem.",
+]
+
+# Shuffling the dataset and delegating 5% to titling tasks.
+# Calculating the number of titling tasks and writing tasks.
+num_rows = len(df)
+num_titling_tasks = int(num_rows * 0.05)
+num_writing_tasks = num_rows - num_titling_tasks
+
+# Shuffle the rows in the DataFrame.
+df = df.sample(frac=1)
+
+# Split the DataFrame into two DataFrames, one for titling tasks and one for writing tasks.
+writing_tasks = df.iloc[:num_writing_tasks]
+titling_tasks = df.iloc[num_writing_tasks:]
+
+prepared_data = []
+
+# Loop through the writing tasks and process them.
+for index, row in writing_tasks.iterrows():
+    # Get data from the entry
+    poem = row["Poem"]
+    topics = row["Tags"]
+    title = row["Title"]
+    author = row["Poet"]
+
+    # Variables to store to instruction, reply, source, and metadata.
+    instruction = random.choice(writing_prompts_topic).replace("$topic", str(topics))
+    reply = random.choice(replies_topic).replace("$topic", str(topics)).replace("$title", title).replace("$poem", poem)
+    source = "PoetryFoundation.org" + " - " + author
+    metadata = {"author": author, "title": title, "tags": str(topics), "task_type": "writing"}
+
+    # If the entry has "nan" as a topic, change the prompts to the non-topic ones.
+    if " nan" in instruction:
```

Comment:
please use `pd.isna(instruction)` or something similar, the "nan" is an artifact from converting NaN (missing/None) to string